builder/vmware/common/driver_workstation9.go

```diff
@@ -173,33 +210,12 @@ func (d *Workstation9Driver) Verify() error {
 	}
 
 	d.VmwareDriver.NetworkMapper = func() (NetworkNameMapper, error) {
-		pathNetmap := workstationNetmapConfPath()
-
-		// Check that the file for the networkmapper configuration exists. If there's no
-		// error, then the file exists and we can proceed to read the configuration out of it.
-		if _, err := os.Stat(pathNetmap); err == nil {
-			log.Printf("Located networkmapper configuration file: %s", pathNetmap)
-			return ReadNetmapConfig(pathNetmap)
-		}
-
-		// If we weren't able to find the networkmapper configuration file, then fall back
-		// to the networking file which might also be in the configuration directory.
-		libpath, _ := workstationVMwareRoot()
-		pathNetworking := filepath.Join(libpath, "networking")
-		if _, err := os.Stat(pathNetworking); err != nil {
-			return nil, fmt.Errorf("error determining network mappings from files in path: %s", libpath)
-		}
-
-		// We were able to successfully stat the file.. So, now we can open a handle to it.
-		log.Printf("Located networking configuration file: %s", pathNetworking)
-		fd, err := os.Open(pathNetworking)
+		// Check if the network mapper configuration file exists.
+		mapper, err := checkNetmapConfExists()
```

Comment:
I like this approach. The consolidation makes this code easier to read. I was surprised to learn that the checkNetmapConfExists also generates. To not make this function do more then what it should I suggest splitting out the logic in the function and making the call here. For example

```go
		mapper, err := checkNetmapConfExists()
		if err == nil {
		    return mapper, nil
		}
		
		if err != nil && runtime.GOOS != "windows" {
		   	// For non-Windows, if neither the default nor alternate
			// configuration paths exist, return an error.
			return nil, err
		}
		if !errors.Is(err, os.ErrNotExist) {
		   return nil, err
		}
		// For windows we can safely generate the network mapper configuration file in the default path.
		pathNetmap, err := generateNetmapConfig()
		if err != nil {
			log.Fatalf("error generating the absent network configuration file: %v", err)
			return nil, err
		}
		// If the file exists, attempt to read the newly generated
		// network mapper configuration file.
		return ReadNetmapConfig(pathNetmap)
```

Reply:
On it - will update and send back soon.

Reply:
Resolved in https://github.com/hashicorp/packer-plugin-vmware/commit/2ebbe20b26e25aa4be8dd12291b20fc408d291ff and tested!